apps/vr-tests-react-components/src/utilities/getStoryVariant.tsx

```diff
@@ -1,23 +1,32 @@
 import * as React from 'react';
 import type { Args, Decorator, StoryFn, StoryObj } from '@storybook/react';
 import { FluentProvider } from '@fluentui/react-provider';
-import { webLightTheme, webDarkTheme, teamsHighContrastTheme } from '@fluentui/react-theme';
+import { webLightTheme, webDarkTheme, teamsHighContrastTheme, type Theme } from '@fluentui/react-theme';
 
 export const DARK_MODE = 'Dark Mode';
 export const HIGH_CONTRAST = 'High Contrast';
 export const RTL = 'RTL';
 
 type StoryVariant = typeof DARK_MODE | typeof HIGH_CONTRAST | typeof RTL;
 
+function isStoryFn(story: StoryFn | StoryObj): story is StoryFn {
+  return typeof story === 'function';
+}
+
+interface StoryObjVariant extends StoryObj {
+  storyName: string;
+  parameters: StoryObj['parameters'] & { dir: ReturnType<typeof getDir>; theme: Theme; mode: 'vr-test' };
+}
+
 /** Helper function that returns RTL, Dark Mode or High Contrast variant of an existing story. */
-export function getStoryVariant(story: StoryFn, variant: StoryVariant) {
+export function getStoryVariant(story: StoryFn | StoryObj, variant: StoryVariant): StoryObjVariant {
```

Comment:
improved types that reflect the actual usage ( we support both CSF2 and CSF3 🥹 )